libs/components/lists/src/lib/modules/sort/sort-item.component.ts

```diff
@@ -30,63 +30,66 @@ export class SkySortItemComponent implements OnInit, OnChanges, OnDestroy {
    * Indicates whether the sorting option is active.
    */
   @Input()
-  public active: boolean;
+  public active: boolean | undefined;
 
   /**
    * Fires when a sort item is selected.
    */
   @Output()
-  public itemSelect: EventEmitter<any> = new EventEmitter();
+  public itemSelect: EventEmitter<any> = new EventEmitter(); // TODO: EventEmitter is of type <void>
 
   public isSelected: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(
     false
   );
 
-  private subscription: Subscription;
+  #subscription!: Subscription;
 
-  private sortItemId: string;
+  #sortItemId!: string;
```

Comment:
`| undefined`

Reply:
```
ngOnInit(): void {
    sortItemIdNumber++;
    this.#sortItemId = SORT_ITEM_ID_PREFIX + sortItemIdNumber.toString();
```

We know that the `sortItemId` will be set in the `ngOnInit()`. Is that not sufficient to assert it as non-null?

Reply:
The property can technically be undefined before the `OnInit` lifecycle hook is called, so we need to ensure we cover the undefined state in our code for those cases. For example, if the property is attempted to be accessed in the constructor, or in another property's getter/setter before `OnInit` is executed.

This same logic applies to `@ViewChild` properties as well. While we can be sure a view child exists in the template when `AfterViewInit` has been called, it's technically undefined beforehand, so we need to account for that in our component.

Reply:
done in [use undefined instead of non-null assertion](https://github.com/blackbaud/skyux/pull/686/commits/3f4900fe29c78136ee1d0a7fb93f641c7d919a2c)